fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/AvroDimensionRowParser.java

```diff
@@ -0,0 +1,192 @@
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+import org.apache.avro.Schema;
+import org.apache.avro.file.DataFileReader;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.io.DatumReader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+/**
+ * Parses an AVRO file into Dimension Rows.
+ */
+public class AvroDimensionRowParser {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AvroDimensionRowParser.class);
+    private ObjectMapper objectMapper;
+
+    private DimensionFieldMapper dimensionFieldMapper;
+
+    /**
+     * Constructs an AvroDimensionRowParser object based on the DimensionFieldMapper object.
+     *
+     * @param dimensionFieldMapper Object that defines the dimension field name transformations
+     * @param objectMapper Object that is used to construct mapper objects
+     */
+    public AvroDimensionRowParser(DimensionFieldMapper dimensionFieldMapper, ObjectMapper objectMapper) {
+        this.dimensionFieldMapper = memoize(dimensionFieldMapper);
+        this.objectMapper = objectMapper;
+    }
+
+    /**
+     * Validates the schema of the given AVRO file with the Dimension schema configured by the user.
+     *
+     * @param dimension The dimension object used to configure the dimension
+     * @param avroSchemaPath The path of the AVRO schema file (.avsc)
+     *
+     * @return true if the schema is valid, false otherwise
+     *
+     * @throws IllegalArgumentException thrown if JSON object `fields` is not present
+     * @throws IOException thrown if there is error parsing the avro files
+     *
+     * The sample schema is expected to be in the following format
+     *
+     * {
+     *    "type" : "record",
+     *    "name" : "TUPLE_0",
+     *    "fields" : [
+     *    {
+     *    "name" : "FOO_ID",
+     *    "type" : [ "null","int" ]
+     *    },
+     *    {
+     *    "name" : "FOO_DESC",
+     *    "type" : [ "null", "string" ]
+     *    }
+     *    ]
+     * }
+     *
+     */
+    public boolean isSchemaValid(Dimension dimension, String avroSchemaPath)
+        throws IllegalArgumentException, IOException {
+
+        // Convert the AVRO schema file into JsonNode Object
+        JsonNode jsonAvroSchema = convertFileToJsonNode(avroSchemaPath);
+
+        // If the JSON schema file is corrupted/ empty, then return the schema is invalid
+        if (jsonAvroSchema == null) {
+            return false;
+        }
+
+        try {
+            jsonAvroSchema = jsonAvroSchema.get("fields");
+        } catch (NullPointerException e) {
```

Comment:
Good catch. This check actually needs to look to see if `jsonAvroSchema.get("fields")` is null, and throw on that, rather than looking for an NPE. Otherwise thNPE will pop out when we call `jsonAvroSchema.spliterator()` later.